notebooks/de_dommel/05_add_berging.py

```diff
@@ -0,0 +1,278 @@
+        area = np.concat([area, window_data[mask].ravel()])
+
+    # Make sure area is never larger than polygon-area
+    area[area > basin_polygon.area] = basin_polygon.area
+
+    # If area is empty, we add min_level at 5% of polygon-area
+    if area.size == 0:
+        level = np.append(level, min_level)
+        area = np.append(area, basin_polygon.area * 0.05)
+
+    # Add extra row with max_level at basin_polygon.area
+    level = np.append(level, max_level)
+    area = np.append(area, basin_polygon.area)
+
+    # In pandas for magic
+    df = pd.DataFrame({"level": np.round(level, decimals=2), "area": np.round(area)})
+    df.sort_values(by="level", inplace=True)
+    df = df.set_index("level").cumsum().reset_index()
+    df.dropna(inplace=True)
+    df.drop_duplicates("level", keep="last", inplace=True)
+
+    # Return profile
+    return basin.Profile(area=df.area, level=df.level)
+
+
+# %% add columns
+
+with rasterio.open(lhm_rasters) as raster_src:
+    # read band
+    maaiveld_data = raster_src.read(banden["maaiveld"])
+
+# sample rasters
+ghg = sample_raster(
+    raster_file=lhm_rasters,
+    df=basin_area_df,
+    band=banden["ghg_2010-2019"],
+    all_touched=True,
+    maaiveld_data=maaiveld_data,
+)
+basin_area_df.loc[:, ["ghg"]] = pd.Series(dtype=float)
+basin_area_df.loc[:, ["ghg"]] = [i["mean"] for i in ghg]
+
+glg = sample_raster(
+    raster_file=lhm_rasters,
+    df=basin_area_df,
+    band=banden["glg_2010-2019"],
+    all_touched=True,
+    maaiveld_data=maaiveld_data,
+)
+basin_area_df.loc[:, ["glg"]] = pd.Series(dtype=float)
+basin_area_df.loc[:, ["glg"]] = [i["mean"] for i in glg]
+
+ma = sample_raster(raster_file=ma_raster, df=basin_area_df, all_touched=True, fill_value=37)  # 37mm/dag is
+basin_area_df.loc[:, ["ma"]] = pd.Series(dtype=float)
+basin_area_df.loc[:, ["ma"]] = [i["mean"] for i in ma]
+
+maaiveld = sample_raster(
+    raster_file=lhm_rasters, df=basin_area_df, band=banden["maaiveld"], stats="mean min max", all_touched=True
+)
+basin_area_df.loc[:, ["maaiveld"]] = pd.Series(dtype=float)
+basin_area_df.loc[:, ["maaiveld"]] = [i["mean"] if pd.isna(i["mean"]) else i["mean"] for i in maaiveld]
+basin_area_df.loc[:, ["maaiveld_max"]] = [i["max"] if pd.isna(i["max"]) else i["max"] for i in maaiveld]
+basin_area_df.loc[:, ["maaiveld_min"]] = [i["min"] if pd.isna(i["min"]) else i["min"] for i in maaiveld]
+# %%update model
+edge_id = model.edge.df.index.max() + 1
+for row in model.basin.node.df.itertuples():
+    # row = next(row for row in model.basin.node.df.itertuples() if row.Index == 1013)
+    node_id = row.Index
+
+    if node_id in basin_area_df.index:
+        # basin-polygon
+        basin_row = basin_area_df.loc[node_id]
+        basin_polygon = basin_area_df.at[node_id, "geometry"]
+
+        # add basin-node
+        basin_node_id = (
+            model.next_node_id
+        )  # FIXME: can be removed if issue is closed https://github.com/Deltares/Ribasim/issues/1805
```

Comment:
Would be good to remove any fixme that are already resolved, otherwise they will linger around forever.

Also, https://github.com/Deltares/Ribasim/issues/1805 seems unrelated?

Reply:
Strange, it's the not properly adding a next node_id issue. Will test all next week when we do new authorities.